op-node/rollup/engine/engine_controller.go

```diff
@@ -376,7 +383,7 @@ func (e *EngineController) InsertUnsafePayload(ctx context.Context, envelope *et
 	if e.syncStatus == syncStatusWillStartEL {
 		b, err := e.engine.L2BlockRefByLabel(ctx, eth.Finalized)
 		rollupGenesisIsFinalized := b.Hash == e.rollupCfg.Genesis.L2.Hash
-		if errors.Is(err, ethereum.NotFound) || rollupGenesisIsFinalized {
+		if errors.Is(err, ethereum.NotFound) || rollupGenesisIsFinalized || e.engineKind != sync.EngineClientGeth {
```

Comment:
> Not something we need to have in this PR at all, but down the line we could do something like || e.engineKind.SyncBehaviorDoesTheThing(), and then all the various engine kinds can define their preferred behavior somewhere else.

I like this idea since we can organize how each behavior is defined by the engine kind. Trying to think of a good naming; what do you think about something like `engineKind.SupportsPostFinalizationELSync` or `engineKind.CanContinueELSync`?

Reply:
between these two, `SupportsPostFinalizationELSync` gives me a greater understanding